Sources/PackageDescription/Product.swift

```diff
@@ -8,46 +8,125 @@
  See http://swift.org/CONTRIBUTORS.txt for Swift project authors
 */
 
-public struct Product {
-    public let name: String
-    public let type: ProductType
-    public let modules: [String]
+/// Defines a product in the package.
+public enum Product {
 
-    public init(name: String, type: ProductType, modules: String...) {
-        self.init(name: name, type: type, modules: modules)
+    /// An exectuable product.
+    public struct ExecutableProduct {
+
+        /// The name of the executable product.
+        public let name: String
+
+        /// The targets in the product.
+        public let targets: [String]
+    }
+
+    /// A library product.
+    public struct LibraryProduct {
+
+        /// The type of library products.
+        public enum LibraryType: String {
+            case `static`
+            case `dynamic`
+        }
+
+        /// The name of the library product.
+        public let name: String
+
+        /// The type of the library.
+        ///
+        /// If the type is unspecified, package manager will automatically choose an optimized type.
+        public let type: LibraryType?
+
+        /// The targets in the product.
```

Comment:
As above, this comment should probably clarify that the array contains the _names_ of the targets.